lib/chatter_web/live/chat_live/index.ex

```diff
@@ -42,24 +58,31 @@ defmodule ChatterWeb.ChatLive.Index do
 
   @impl true
   def handle_event("search", %{"search" => params}, socket) do
-    with {:ok, search} <- Search.create(params) do
+    with {:ok, search} <- Search.create(params),
+      search_messages <- user_shown_messages({search, socket}) do
```

Comment:
do you need full socket here ?

Reply:
no, changed `user_shown_messages` params  to {search, filter_option}